### PR TITLE
Initial version proposal

### PR DESCRIPTION
**Pull request by code-specialist**
## Done
- Initial version proposal
- Docs: https://fastapi-keycloak.code-specialist.com/

## TODO before merging

- [x] Unit tests
<s>- [ ] 100% test coverage</s>
<s>- [ ] Test infrastructure running on actions runner</s>

### CODE REVIEW

File: .github/workflows/testing.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+#          python -m pip install pytest
+#          python -m pip install pytest-cov
+#
+#      - name: "Install dependencies"
+#        run: |
+#          python -m pip install -r requirements.txt
+#
+#      - name: Test with pytest
+#        run: |
+#          pytest tests
```
Comment (code-specialist) [inline .github/workflows/testing.yaml:34]: Testing works locally only, currently.

File: mkdocs.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
+site_name: FastAPI Keycloak Integration
+site_url: https://docs.tebuto.com
```
Comment (code-specialist) [inline mkdocs.yaml:2]: Todo

File: mkdocs.yaml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        icon: material/toggle-switch
+        name: Switch to light mode
+  font:
+    text: Nunito Sans
+    code: IBM Plex Mono
+
+nav:
+  - Introduction: ./index.md
+  - Installation: ./installation.md
+  - Quickstart: ./quick_start.md
```
Comment (code-specialist) [inline mkdocs.yaml:35]: Add example config

File: fastapi_keycloak/__init__.py [hunk 4]
```diff
@@ -0,0 +1,4 @@
+from fastapi_keycloak.api import FastAPIKeycloak
+from fastapi_keycloak.model import OIDCUser, UsernamePassword, HTTPMethod
+
+__all__ = [FastAPIKeycloak, OIDCUser, UsernamePassword, HTTPMethod]
```
Comment (JonasScholl) [inline fastapi_keycloak/__init__.py:4]: `__all__` has to contain stings (the names of the variables), so I would suggest to use the `__name__` property of the classes

File: fastapi_keycloak/setup.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    long_description=read_description(),
+    long_description_content_type="text/markdown",
+    url='https://github.com/code-specialist/fastapi-keycloak',
+    install_requires=read_dependencies(),
+    python_requires='>=3.8',
+    classifiers=[
+        "Programming Language :: Python :: 3",
+        "Operating System :: OS Independent",
+    ],
+)
```
Comment (JonasScholl) [inline fastapi_keycloak/setup.py:38]: When we publish this package on PyPI, we an and should add some more information, but we can do this later in another PR, just wanted to mention it

**Review by JonasScholl — APPROVED** [review]
Lgtm, looks like a lot of effort 😅👍 I think we should also focus on good and readable tests, especially when we open-source it, everybody should be able to understand them

**Comment by JonasScholl** [comment]
Btw, let's have a discussion about die the naming / categorization of tests :D I think the naming here doesn't fit to the type of tests. For instance: functional tests are a kind of black box tests where you call a system or a (bigger) system component and expect a specific output. You don't care about what happens inside. The word "funcional" is realated to the specified functional requirements and not meant as a function in the source code. The current tests in the functional-tests folder are clearly unit tests

File: fastapi_keycloak/setup.py (hunk 5 above)
Comment (code-specialist) [reply fastapi_keycloak/setup.py:38]: What kind of information would you add?

**Comment by code-specialist** [comment]
> Btw, let's have a discussion about die the naming / categorization of tests :D I think the naming here doesn't fit to the type of tests. For instance: functional tests are a kind of black box tests where you call a system or a (bigger) system component and expect a specific output. You don't care about what happens inside. The word "funcional" is realated to the specified functional requirements and not meant as a function in the source code. The current tests in the functional-tests folder are clearly unit tests in my opinion 😄

For me, a unit is a collection of functions, a class so to say, that fully sustains itself. A functional test tests functional behavior. I'm aware this is not common sense and that my personal interpretation of these two terms might irritate any logical thinking human being which is why I will rename them correspondingly

**Pull request merged**